### PR TITLE
Add support for higher DS versions

### PR DESCRIPTION
**Pull request by laeubi**
Currently PDE only declares versions up to DS 1.3 in the model but there are already 1.4/1.5 out.

Even though not all features are supported we should allow the user to select a higher version (e.g. to use with Tycho) and to make them aware other versions exits and might can need some improvements/contributions.

### CODE REVIEW

File: ds/org.eclipse.pde.ds.tck/src/org/eclipse/pde/ds/tck/Tck1.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.eclipse.pde.ds.tck;
```
Comment (akurtakov) [inline ds/org.eclipse.pde.ds.tck/src/org/eclipse/pde/ds/tck/Tck1.java:1]: Make sure that you put license headers to every new file.

**Comment by laeubi** [comment]
Current status is that I got it working that PDE is able to handel things **on the validation level** except for this what is probably a bug/something to improve in the TCK itself:

- https://github.com/osgi/osgi/issues/636

so currently there are two validation problems reported:
```
tck/OSGI-INF/impl-src/org/osgi/impl/bundle/component/annotations/ServiceBundle.java:30 Property 'servicefactory' is ignored when non-default scope is specified. 
tck/OSGI-INF/impl-src/org/osgi/impl/bundle/component/annotations/ServiceSingleton.java:30 Property 'servicefactory' is ignored when non-default scope is specified. 
```

beside that, PDE-DS has 28 passing and 10 tests left that do not pass not that bad for a first run:
```
Test run finished after 836 ms
[         5 containers found      ]
[         0 containers skipped    ]
[         5 containers started    ]
[         0 containers aborted    ]
[         5 containers successful ]
[         0 containers failed     ]
[        38 tests found           ]
[         0 tests skipped         ]
[        38 tests started         ]
[         0 tests aborted         ]
[        28 tests successful      ]
[        10 tests failed          ]
```

**Comment by laeubi** [comment]
Got it down to 8 testfailures now, things now complained about is mostly properties and properties ordering issues.

**Comment by laeubi** [comment]
This now passes all the OSGi TCKs!!! :1st_place_medal:

**Comment by HannesWell** [comment]
> This now passes all the OSGi TCKs!!! 🥇

This is awesome.

Can you please explain how this relates to your other two DS related PRs?
- https://github.com/eclipse-pde/eclipse.pde/pull/926
- https://github.com/eclipse-pde/eclipse.pde/pull/932

**Comment by laeubi** [comment]
N&N added here https://github.com/eclipse-platform/www.eclipse.org-eclipse/pull/98

**Pull request merged**